inst/tests/test-parsers.R

```diff
@@ -156,6 +156,20 @@ test_that("ymd functions correctly parse dates with no separators and no quotes"
               equals(as.POSIXct("2010-01-02 23:59:59", tz = "UTC")))
 })
 
+test_that("numbers are correctly converted into character for parsing", {
+  # numbers with 000000 at the end are wrongly converted by as.character
+  # which outputs scientific notation
+  expect_that(ymd_hms(20100102000000),
+              equals(as.POSIXct("2010-01-02 00:00:00", tz = "UTC")))
+  expect_that(.num_to_date(20100102000000),
+              equals("20100102000000"))
+  # check for a fix in as.character
```

Comment:
I don't think we need these tests.  `as.character` is not broken and will never be fixed.